cng/tls1prf.go

```diff
@@ -73,11 +76,15 @@ func TLS1PRF(secret, label, seed []byte, keyLen int, h func() hash.Hash) ([]byte
 		Count:   uint32(len(buffers)),
 		Buffers: &buffers[0],
 	}
-	out := make([]byte, keyLen)
 	var size uint32
-	err = bcrypt.KeyDerivation(kh, params, out, &size, 0)
+	err = bcrypt.KeyDerivation(kh, params, result, &size, 0)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return out[:size], nil
+	// The Go standard library expects TLS1PRF to return the requested number of bytes,
+	// fail if it doesn't.
+	if size != uint32(len(result)) {
```

Comment:
This is just a sanity check, right? Or does the caller need to be careful to pass in a slice with a valid size?

Ironically, without this comment, I'd assume this is just double-checking CNG's work, but with it, I think it should probably also mention when this error should be expected. (Even if it's as simple as "CNG didn't do what we asked it to".) 😄

Reply:
It's a sanity check. `BCryptKeyDerivation` is a generic function that works with multiple algorithms, some of them can return less bytes than requested. It shouldn't be the case for TLS1PRF, and I haven't been able to trigger that, but I've added this check just for my own sanity 😄.

Reply:
Maybe it's worth including:

```suggestion
	// The Go standard library expects TLS1PRF to return the requested number of bytes,
	// fail if it doesn't. While there is no known situation where this will happen,
	// BCryptKeyDerivation handles multiple algorithms and there could be a subtle mismatch
	// after more code changes in the future.
	if size != uint32(len(result)) {
```